ui/shared/Clipboard.jsx

```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faClipboard } from '@fortawesome/free-regular-svg-icons';
+
+const Clipboard = ({ description, text }) => {
+  if (!text) {
+    return null;
+  }
+
+  const copyToClipboard = () => navigator.clipboard.writeText(text);
+
+  return (
+    <button
+      type="button"
+      title={`Copy ${description}`}
+      onClick={copyToClipboard}
+      className="px-1 mx-1"
+    >
+      <FontAwesomeIcon icon={faClipboard} />
+    </button>
+  );
+};
+
+Clipboard.propTypes = {
+  description: PropTypes.string.isRequired,
+  text: PropTypes.string,
```

Comment:
This implementation is ok, but we typically don't return null if a prop type is falsey - we'd just make `text` a required prop type and then add the condition for rendering the Clipboard component where it's used.